jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarToJibContainerBuilderConverter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.jar;
+
+import com.google.cloud.tools.jib.api.InvalidImageReferenceException;
+import com.google.cloud.tools.jib.api.Jib;
+import com.google.cloud.tools.jib.api.JibContainerBuilder;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+/** Class to build a container representation from the contents of a jar file. */
+public class JarToJibContainerBuilderConverter {
+
+  /**
+   * Generates a {@link JibContainerBuilder} from contents of a jar file.
+   *
+   * @param jarPath path to the jar file
+   * @param tempDirPath path to a temporary directory which will be used store the exploded jar's
+   *     contents
+   * @return JibContainerBuilder
+   * @throws IOException if I/O error occurs when opening the jar file or if temporary directory
+   *     provided doesn't exist
+   * @throws InvalidImageReferenceException if the base image used is invalid
+   */
+  public static JibContainerBuilder toJibContainerBuilder(Path jarPath, Path tempDirPath)
```

Comment:
Not always, `to<AnotherClass>()` is conventional for an _instance_ of a class to transform to or provide an instance of another class. And in this case, the method name can be made more clear. Or, maybe, this is a very short code block, and creating a converter class is an overkill? But honestly, I don't grasp the overall design honestly, I am not sure.

And, perhaps this can be package-private? Let's start with limited scope. It's easy to make it `public`, but it's difficult to find methods with unnecessarily larger access.

Reply:
I agree, the naming was a little unclear. I modified the class name to be a bit more consistent with `BuildFiles` so that the method will be called in this format -> `JarFiles.toJibContainerBuilder.` But let me know if this looks okay? The method is very small now, but the plan is to eventually pass in `JibCli` options as another parameter, which will be used to set those extra image parameters (such as environment, etc.). 

Ah since this method is going to be accessed in the `com...jib.cli.cli2` package when we are containerizing the application after the jar subcommand is called, I decided to make it public. Do you think this change is a little premature? Or in other words, is it better to increase the scope only when we actually use it elsewhere.

Reply:
Hmm... what if this methods goes into `JarProcessor`? Do you have a specific design principle in mind that you wanted to separate it out?

And if it is certain that this will be called from a different package, we can leave it as `public`. I just wasn't sure and thought the possibility this may be called in the same package.

Reply:
That is an interesting idea. I could be thinking about this too early but I was slightly reluctant on making `JarProcessor` do too many things at once. Additionally, since we're also planning on adding more modes in the future, this would result in the class become pretty long - so I decided to have one class that does just the processing of the jar and another one that does the generating of `JibContainerBuilder` (and incorporating cli params in the future)
I totally understand that the class may seem a little small right now but once we have more modes + jar types, this class would probably need to do the following:
1. Get the mode from JibCli options (default would be exploded mode).
2. Call `JarProcessor.determineJarType` to determine the jar type. 
3. Have some logic as to which `JarProcessor` processing method to call (either explodeJar or packageJar) based on the mode and the type of jar.
4. [In the future] Incorporate optional cli parameters.

Reply:
That makes sense. I just felt that the name `JarFiles` is too generic and easily misunderstood as a general JAR file utility, where this is really specific to instantiate and configure `JibContainerBuilder`. I really wanted to avoid `JarFiles` and tried to think of a better name, and then got to the idea of putting it into `JarProcessor`. However, even after thinking about the name for a long time, I can't really suggest something good. Otherwise, LGTM.

Reply:
And I guess the name will really depend on what this class will do going forward and what methods it will have at the end.

Reply:
That is true, `JarFiles` is not the best name. How does `JarToJibContainerBuilderPopulator` sound?

Reply:
If the class is for configuring `JibContainerBuilder` (and only for it), I think `JarProcessor.computeEntrypointForExplodedStandard()` should be in this class. I'm a bit confused with the boundary between this class and `JarProcessor`. Maybe after moving `computeEntrypointForExplodedStandard()`, rename `JarProcessor` to `LayerPopulator` (the class seems only specific to create layers for different modes) and rename this class to `JarProcessor`? This class seems to be more general than "LayerPopulator", where the latter is at a more lower level specific to creating layers used by this class.

Reply:
As discussed offline, since the methods to compute the entrypoint also depend on the modes of processing the jar, will keep these methods as they are and rename `JarProcessor` to `JarModeProcessor` (since this class focuses on processing the jar based on particular modes).